### ISSUE TITLE
Inplace operation: error

### ISSUE THREAD

**Issue by abril4416** [iesl/diora#7]
For the line 212 in diora/net/diora.py, you defined a function named inside_fill_chart, it fill the latest h,c,s to the chart. This is actually an inplace operation (you directly modified values of elements in the matrix). However, the matrix, chart.inside_h, chart_inside_c and chart.inside_s are all used for gradient back propagation. I wonder which pytorch version you used. Since the version >=0.4.0 does not allow such operation.

**Comment by mrdrozdov** [comment]
For in-place operations like this, I believe that pytorch keeps an internal count on how many times the tensor has been modified.

We allocate a large tensor to cache many of the intermediate values (this is especially useful when selecting / slicing tensors used during dynamic programming), which is the same large tensor modified by `inside_fill_chart`, `outside_fill_chart`, etc. Since we never overwrite values, backprop will work as expected. If we did overwrite values, pytorch should throw an exception even in latest versions.
